src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/AuroraTopologyService.java

```diff
@@ -215,22 +217,36 @@ protected ClusterTopologyInfo queryForTopology(JdbcConnection conn) throws SQLEx
     long startTimeMs = System.currentTimeMillis();
 
     ClusterTopologyInfo topologyInfo = null;
-
-    try (Statement stmt = conn.createStatement()) {
-      try (ResultSet resultSet = stmt.executeQuery(RETRIEVE_TOPOLOGY_SQL)) {
-        topologyInfo = processQueryResults(resultSet);
-      }
-    } catch (SQLSyntaxErrorException e) {
-      // We may get SQLSyntaxErrorException like the following from MySQL databases:
-      // "Unknown table 'REPLICA_HOST_STATUS' in information_schema"
-      // Ignore this kind of exceptions.
+    try {
+      do {
+        try (Statement stmt = conn.createStatement()) {
+          try (ResultSet resultSet = stmt.executeQuery(RETRIEVE_TOPOLOGY_SQL)) {
+            topologyInfo = processQueryResults(resultSet);
+          }
+        } catch (SQLSyntaxErrorException e) {
+          // We may get SQLSyntaxErrorException like the following from MySQL databases:
+          // "Unknown table 'REPLICA_HOST_STATUS' in information_schema"
+          // Ignore this kind of exceptions.
+        }
+        if(initMultiWriterCluster != null && !initMultiWriterCluster
+                && topologyInfo != null && topologyInfo.getMultiWriterCluster()){//When failover, MultiWriterCluster returns true, and restore after re-query
+          TimeUnit.SECONDS.sleep(2L);
+        }
+      } while (initMultiWriterCluster != null && !initMultiWriterCluster
+              && topologyInfo != null && topologyInfo.getMultiWriterCluster());
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
     } finally {
       if (gatherPerfMetrics(conn.getPropertySet())) {
         long currentTimeMs = System.currentTimeMillis();
         this.metricsContainer.registerTopologyQueryExecutionTime(currentTimeMs - startTimeMs);
       }
     }
-
+    Optional.ofNullable(topologyInfo).ifPresent(e->{
+      if(initMultiWriterCluster==null) {//The initial value setting must be correct
+        initMultiWriterCluster = e.getMultiWriterCluster();
+      }
+    });
```

Comment:
```suggestion

    Optional.ofNullable(topologyInfo).ifPresent(e -> {
      //The initial value setting must be correct
      if (initMultiWriterCluster == null) {
        initMultiWriterCluster = e.getMultiWriterCluster();
      }
    });
    
```
Can the spacing be changed here too?

Reply:
The annotation position has been modified